lib/portlayer/exec/exec.go

```diff
@@ -76,12 +80,47 @@ func Init(ctx context.Context, sess *session.Session, source extraconfig.DataSou
 			Config.ResourcePool = o.ResourcePool
 		case *object.ResourcePool:
 			Config.ResourcePool = o
+			Config.VMGroupName = o.Name()
+			rp := compute.NewResourcePool(ctx, sess, cr)
+			Config.Cluster, err = rp.GetCluster(ctx)
+			if err != nil {
+				err = fmt.Errorf("could not get cluster from resource pool: %s", err)
+				return
+			}
+
 		default:
 			err = fmt.Errorf("could not get resource pool or virtual app from reference %q: object type is wrong", cr.String())
 			log.Error(err)
 			return
 		}
 
+		// stash this aside for future use in vm group manipulation
+		Config.SelfReference = self
+
+		// create the VM group for DRS rule use
+		var clusterConfig mo.ClusterComputeResource
+		vmGroupOpType := types.ArrayUpdateOperationAdd
+
+		err = Config.Cluster.Properties(ctx, Config.Cluster.Reference(), []string{"configurationEx"}, &clusterConfig)
+		if err != nil {
+			log.Errorf("Unable to obtain cluster config: %s", err)
+			return
+		}
+
+		clusterConfigEx := clusterConfig.ConfigurationEx.(*types.ClusterConfigInfoEx)
```

Comment:
clusterConfig is a zero initialized variable. so next for-loop basically doesn't make any sense since iterating over empty array is not going to happen. Am I missing something?

Reply:
yes, I am. &clusterConfig reference is on the next line.